### ISSUE TITLE
Compilation fails when path is a template literal

### ISSUE THREAD

**Issue by bernardmcmanus** [thgreasi/babel-plugin-system-import-transformer#10]
Runing:
- `babel-core@6.14.0`
- `babel-plugin-system-import-transformer@2.3.0`

``` js
// this is ok
System.import('./pages/' + name + '/route');

// this is not
System.import(`./pages/${name}/route`);
```

Adding `transform-es2015-template-literals` to plugins before `system-import-transformer` does not seem to have any effect. Here is the full stack trace:

```
TypeError: /var/app/routes.js: Property value expected type of string but got null
    at Object.validate (/var/app/node_modules/babel-types/lib/definitions/index.js:161:13)
    at validate (/var/app/node_modules/babel-types/lib/index.js:505:9)
    at Object.builder (/var/app/node_modules/babel-types/lib/index.js:466:7)
    at new SystemImportExpressionTransformer (/var/app/node_modules/babel-plugin-system-import-transformer/dist/SystemImportExpressionTransformer.js:28:63)
    at PluginPass.CallExpression (/var/app/node_modules/babel-plugin-system-import-transformer/dist/index.js:16:31)
    at newFn (/var/app/node_modules/babel-traverse/lib/visitors.js:276:21)
    at NodePath._call (/var/app/node_modules/babel-traverse/lib/path/context.js:76:18)
    at NodePath.call (/var/app/node_modules/babel-traverse/lib/path/context.js:48:17)
    at NodePath.visit (/var/app/node_modules/babel-traverse/lib/path/context.js:105:12)
    at TraversalContext.visitQueue (/var/app/node_modules/babel-traverse/lib/context.js:150:16)
    at TraversalContext.visitSingle (/var/app/node_modules/babel-traverse/lib/context.js:108:19)
    at TraversalContext.visit (/var/app/node_modules/babel-traverse/lib/context.js:192:19)
    at Function.traverse.node (/var/app/node_modules/babel-traverse/lib/index.js:114:17)
    at NodePath.visit (/var/app/node_modules/babel-traverse/lib/path/context.js:115:19)
    at TraversalContext.visitQueue (/var/app/node_modules/babel-traverse/lib/context.js:150:16)
    at TraversalContext.visitMultiple (/var/app/node_modules/babel-traverse/lib/context.js:103:17)
```

**Comment by thgreasi** [comment]
Actually only string liberals are currently supported as an argument at the
moment. Variables #4, string concatenation expressions and template
liberals are not supported. That's because CommonJS bundlers, most namely
webpack and browserify, analyze the code and require a string literal to be
used, so that they can determine the imported modules during the
compilation step (at least that was how things worked when I implemented
this).

Which bundling system are you using?
If this CommonJS limition doesn't exist any more then we should try adding
support for this. At the moment we should be able to add support for this
for AMD and Global transformation targets.

> ## 
> 
> _Thodoris Greasidis_

**Comment by bernardmcmanus** [comment]
@thgreasi I'm transpiling to CommonJS. As a workaround I'm just resolving a promise with a `require` call, which seems to be working just fine:

``` js
Bluebird.resolve(require(`./pages/${name}/route`))
```

I'm using `webpack@2.1.0-beta.19` so I'm not sure if this is supported in webpack 1, but it recognizes that I'm requiring based on an expression and is using a regex to resolve those imports:

```
 [268] ./app/pages ^\.\/.*\/route$ 239 bytes {0} [built]
    + 645 hidden modules
```

**Comment by thgreasi** [comment]
Thanks for the details. I will check it out as soon as I find some time.
Expressions were not supported back in webpack 1 so I never bothered. Of
course this isn't going to be working for browserify users, which as far as

## I know need string literals as require parameters.

_Thodoris Greasidis_

**Comment by thgreasi** [comment]
Just released v2.4.0 that should try to transform any expression.
Until now we checked if it was an actual string literal.
Of course the result might not work on all environments, but your webpack2 use case should now work.
Have fun (hopefully) and please, please, please, report back whether it worked for you.

**Comment by bernardmcmanus** [comment]
@thgreasi awesome thanks! It's working swimmingly 👍
